lib/client/client.go

```diff
@@ -101,8 +104,28 @@ func (c *Client) BGPPeers() BGPPeerInterface {
 }
 
 // IPAM returns an interface for managing IP address assignment and releasing.
-func (c *Client) IPAM() IPAMInterface {
-	return newIPAM(c)
+func (c *Client) IPAM() ipam.Interface {
+	return ipam.NewIPAM(c.Backend, poolAccessor{})
+}
+
+type poolAccessor struct {
+	client *Client
+}
+
+func (p poolAccessor) GetEnabledPools(ipVersion int) ([]net.IPNet, error) {
```

Comment:
can you clarify why is this in v1 client?

Reply:
for compilation purposes.  also it allowed me to test a subset of the changes using etcdv2.  I am expecting the v1 client to be removed eventually, so I wouldn't dwell on this too much at this point.